akka-diagnostics/src/test/java/akka/diagnostics/ConfigCheckerSpec.scala

```diff
@@ -0,0 +1,1089 @@
+
+      val checker2 = new ConfigChecker(extSys, config2, reference)
+      val warnings2 = checker2.check().warnings
+
+      printDocWarnings(warnings2)
+      assertCheckerKey(warnings2, "hostname")
+      assertPath(warnings2, "akka.remote.artery.canonical.hostname")
+
+    }
+
+    "not warn about HTTP server, client and pool specific parsing overrides" in {
+      // these are brought in through some trix in akka-http
+      val config = ConfigFactory
+        .parseString("""
+          akka.http.server.parsing.illegal-header-warnings = on
+          akka.http.client.parsing.illegal-header-warnings = on
+          akka.http.host-connection-pool.parsing.illegal-header-warnings = on
+        """)
+        .withFallback(reference)
+
+      val checker = new ConfigChecker(extSys, config, reference)
+      val warnings = checker.check().warnings
+
+      warnings should be(Vector.empty)
+    }
+
+    "not warn about things in the cinnamon config" in {
+      val config = ConfigFactory.parseResources("all-cinnamon-2018-05-14-reference.conf").withFallback(reference)
+
+      val checker = new ConfigChecker(extSys, config, reference)
+      val warnings = checker.check().warnings
+
+      warnings should be(Vector.empty)
+    }
+
+    @nowarn("msg=possible missing interpolator")
+    val defaultGrpc = ConfigFactory.parseString("""
+      akka.grpc.client."*" {
+        host = ""
+        port = 0
+        ssl-config = ${ssl-config}
+      }
+    """)
+
+    "not warn about paths with quotes" in {
+      val config = ConfigFactory
+        .parseString("""akka.grpc.client."*".ssl-config.hostnameVerifierClass = 37""")
+        .withFallback(reference)
+      val referenceWithGrpc = reference.withFallback(defaultGrpc).resolve
+      val checker = new ConfigChecker(extSys, config, referenceWithGrpc)
+      val warnings = checker.check().warnings
+
+      warnings should be(Vector.empty)
+    }
+
+    "warn about paths under akka.grpc.client for specific services" in {
+      val config = ConfigFactory
+        .parseString("""
+        akka.grpc.client {
+          "helloworld.GreeterService" {
+            host = 127.0.0.1
+            poort = 8080
+          }
+          otherValue = 42
+        }
+      """)
+        .withFallback(reference)
+      val referenceWithGrpc = reference.withFallback(defaultGrpc).resolve
+      val checker = new ConfigChecker(extSys, config, referenceWithGrpc)
+      val warnings = checker.check().warnings
+
+      warnings should be(
+        Vector(
+          ConfigWarning(
+            "typo",
+            """akka.grpc.client."helloworld.GreeterService".poort is not an Akka configuration setting. Did you mean one of 'akka.remote.classic.netty.tcp.port', 'akka.remote.classic.netty.ssl.port', 'akka.remote.artery.bind.port'? Is it a typo or is it placed in the wrong section? Application specific properties should be placed outside the "akka" config tree.""",
```

Comment:
Levenshtein distance outputs the following:

```
(port,akka.remote.classic.netty.tcp.port,1)
(port,akka.remote.classic.netty.ssl.port,1)
(port,akka.remote.artery.bind.port,1)
(port,akka.remote.artery.canonical.port,1)
(port,akka.management.http.port,1)
(port,akka.http.client.proxy.https.port,1)
(port,akka.grpc.client.*.port,1)
```

so the three suggestions picked are the first three, while in the original `akka-enhancements` were the last 3. I changed the expectation in this test but I'm not sure this is the right solution. 
Shall we modify the algorithm to make sure the last three are picked as the three suggestions?